src/rgw/rgw_reshard.cc

```diff
@@ -529,14 +529,14 @@ int RGWBucketReshard::do_reshard(int num_shards,
 
   verbose = verbose && (formatter != nullptr);
 
-  if (verbose) {
+  if (verbose && out) {
```

Comment:
would it work to just change the verbose assignment above?
```diff
-  verbose = verbose && (formatter != nullptr);
+  verbose = verbose && (formatter != nullptr) && (out != nullptr);
```

Reply:
I don't particularly like using "verbose" as an alias for "have formatter"

Reply:
Yes, changed to:
https://github.com/ceph/ceph/blob/4221452e90e01d5f27cf3ccf92d0b3a359218c30/src/rgw/rgw_reshard.cc#L530